bootstrap/config/manager/overlays/ocp/kustomization.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- ../../base
+
+patches:
+  - target:
+      kind: Deployment
+      name: controller-manager
+    patch: |-
+      - op: replace
+        path: "/spec/template/spec/containers/0/env/0"
+        value:
+          name: USE_INTERNAL_IMAGE_URL
```

Comment:
I thought this was meant to allow someone deploying this to change the namespace for assisted. How does this do that?

Reply:
This stanza replaces the first environment variable listed in the base deployment yaml with the `value` specified here. 

So the result of `$ kustomize build bootstrap/config/manager/overlays/ocp`
is this:
```yaml
...
apiVersion: apps/v1
kind: Deployment
metadata:
...
  name: controller-manager
  namespace: system
spec:
  replicas: 1
  selector:
    matchLabels:
      control-plane: controller-manager
  template:
    metadata:
      annotations:
        kubectl.kubernetes.io/default-container: manager
      labels:
        control-plane: controller-manager
    spec:
      containers:
      - args:
        - --leader-elect
        command:
        - /manager
        env:
        - name: USE_INTERNAL_IMAGE_URL
          value: "false"
        - name: IMAGE_SERVICE_NAME
          value: assisted-image-service
        - name: IMAGE_SERVICE_NAMESPACE
          value: assisted-installer
        image: quay.io/edge-infrastructure/openshift-capi-agent-bootstrap:latest
        imagePullPolicy: Always
        livenessProbe:
...
```

Whereas just doing `$ kustomize build bootstrap/config/manager/base`
produces this:
```yaml      
...
apiVersion: apps/v1
kind: Deployment
metadata:
...
  name: controller-manager
  namespace: system
spec:
  replicas: 1
  selector:
    matchLabels:
      control-plane: controller-manager
  template:
    metadata:
      annotations:
        kubectl.kubernetes.io/default-container: manager
      labels:
        control-plane: controller-manager
    spec:
      containers:
      - args:
        - --leader-elect
        command:
        - /manager
        env:
        - name: USE_INTERNAL_IMAGE_URL
          value: "true"
        - name: IMAGE_SERVICE_NAME
          value: assisted-image-service
        - name: IMAGE_SERVICE_NAMESPACE
          value: assisted-installer
        image: quay.io/edge-infrastructure/openshift-capi-agent-bootstrap:latest
        imagePullPolicy: Always
        livenessProbe:
...
```

Reply:
For the purpose of the sylva integration, the released manifests are used. For example if release 0.1.5 will have this fix, it's the github.com/openshift-assisted/cluster-api-agent/releases/download/v0.1.5/{bootstrap-components.yaml,controlplane-components.yaml} will be used by Sylva. Since bootstrap-components.yaml is already updated, so I think we are good. We can always overwrite  USE_INTERNAL_IMAGE_URL from  a sylva unit via kustomization.

@CrystalChun This base/ and overlays/ocp/ folder are meant for internal use only, right? maybe they can be re-organized for some e2e CI test and used to deploy all manifests required for the CAPI provider to emulate how an end user will deploy the manifests on a k8s or OCP cluster. But that's outside of the scope of this PR and we can discuss that later.

Reply:
I think the intention of kustomize folder is for users who has cloned the source repo to be able to use kubectl apply -k to deploy the capi pods and resources in one step. The root level README.md does mention kubectl apply -k config/samples/. So I guess the purpose of the base/ and overlays/ folder here will be eventually used for that purpose? @CrystalChun @rccrdpccl